model_compression_toolkit/qunatizers_infrastructure/keras/quantize_wrapper.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================f
+
+
+from model_compression_toolkit.core.common.constants import FOUND_TF
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.qunatizers_infrastructure.common.node_quantization_dispatcher import \
+    NodeQuantizationDispatcher
+
+if FOUND_TF:
+    import tensorflow as tf
+    from tensorflow.python.util import tf_inspect
+    from tensorflow_model_optimization.python.core.keras import utils
+    from keras.utils import deserialize_keras_object, serialize_keras_object
+
+    DISPATCHER = "dispatcher"
+    LAYER = "layer"
+    STEPS = "optimizer_step"
+    TRAINING = "training"
+
+
+    def _make_quantizer_fn(quantizer, x, training):
+        """Use currying to return True/False specialized fns to the cond."""
+
+        def quantizer_fn():
+            return quantizer(x, training)
+
+        return quantizer_fn
+
+
+    def _weight_name(name):
+        """Extracts the weight name from the full TensorFlow variable name.
+
+        For example, returns 'kernel' for 'dense_2/kernel:0'.
+
+        Args:
+          name: TensorFlow variable name.
+
+        Returns:
+          Extracted weight name.
+        """
+        return name.split(':')[0].split('/')[-1]
```

Comment:
I think we use this string splitting in other places of the code, so maybe create some generic helper method instead of duplicating it.